File: ice0208/hash-map_set/1207.py
class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        map = dict()
        for num in arr:
            map[num] = map.get(num, 0) + 1

        cnt_set = set()
        for k, v in map.items():
            cnt_set.add(v)

        return len(map) == len(cnt_set)


if __name__ == "__main__":
    print(Solution().uniqueOccurrences(arr=[1, 2, 2, 1, 1, 3]))
    print(Solution().uniqueOccurrences(arr=[1, 2]))
    print(Solution().uniqueOccurrences(arr=[-3, 0, 1, -3, 1, 1, 1, -3, 10, 0]))

File: moonpoison/hash_map_set/1207.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) 
    {
        map<int, int> m;
        int a[1001]={0};
        for(int i=0;i<arr.size();++i)
            m[arr[i]]++;
        for(auto i=m.begin();i!=m.end();++i)
        {
            a[i->second]++;
            if(a[i->second]>1) return false;
        }
        return true;
    }
};